### PR TITLE
Clarify single union discriminated union declaration.

### LINKED ISSUE #23581: Clarify discriminated union with a single declaration
Hello,

All the examples in the section ["Formatting discriminated union declarations"](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-discriminated-union-declarations) have multiple declarations.
I'd like to know if anything changes when the DU only has one declaration.
In some scenarios, you could omit the bar character.

```fsharp
[<NoEquality; NoComparison>]
type SynBinding =
    | SynBinding of
        accessibility: SynAccess option *
        kind: SynBindingKind *
        mustInline: bool *
        isMutable: bool *
        attributes: SynAttributes *
        xmlDoc: PreXmlDoc *
        valData: SynValData *
        headPat: SynPat *
        returnInfo: SynBindingReturnInfo option *
        expr: SynExpr  *
        range: range *
        seqPoint: DebugPointAtBinding
```

Removing `| ` produces the same result.

I propose to always have the bar because in some scenarios you cannot leave it out:
* When followed by a compiler define (https://github.com/fsprojects/fantomas/issues/1483)
* When it contains XML comment (https://github.com/fsprojects/fantomas/issues/1563)
* Combination of an access modifier with a tuple on a single line ([sample](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAXAngB0WAYrLGALwQA6ATuQHYQTaUCWAbgIbp4A%2BBRYsAMzBMa6MACowAIyIAbMAHoF0tpWEBnMDUTRE69asyQANOCjrE6dCIDm61GADaZiKHr1IAEiy4HkEegmLh6e7LIArshozu7uAAymsR4AkjRIogDKTABeyIlJYAAswRAAusEAvvmuJVDeOFFQAUFJXmGRDjEFAIzVsZAAsmwAHgAyIoijiDQ26AAWLT0ATHG15e5VwW799b5okDKwsovubWwRjV1JS32nGYgAtkxwsrA0AILoAKJpAPIC420J1iAnOFjWlT621Ou0aBzkwIgZwunVqEAAzLcPBlsGxdAAhRACWCURAABVUbAelkQlER9HQlEiEI2UNqXh8cMOxyxyI60TRRSxSJxeMQhOJpNGsAA7rToGwLKkWLAFdY3vSIIzmUl1vRNu5oSFOX5ufS%2BZdBQBWYVQUUEokkxAAVWwuEoCqVNBVaqYGttoNk4N1kK27NhpoRvNC535TkFADZbZB7eLHaSAMKyRXqDNvdTa6DoEmasCB4OxPUQA30I1IiP7M3R9qWgpgADsydTEqdA0eUlppfLiBZ%2BrZrQbUCb4ZbqLbAA4u7iHZLEHnXjQh2CRyHWWGJybG1GZ7HWwUAJxLsXvAScSh5h4PNil7U7yuhw3hw9T48T2cCttugSQUU2XNNV3uJ4Xn9QVh1Hatxx2b94SOc0YxRACel6EDUxvO9IOeI4YLbV94LAGsagPBpI1Q5tTznZZk1SdJ0F%2BGgABVKEwAB1Jh5i3IM33cKtyMQmFkOnP96MwpJukxHCwPeShYHCNIABFEFkJgnjvF8mSE%2BgRIosA6zqZDmjojD40AwpkyGYZkgEdi5mmL5BIyOYSXQXj4H421ClWXcx33JDqP2CyTysq5Ym6G0QPs1IBCYYZfndDgSS%2BYZGD0dQ-U3W0rUC9890-Ki9iaUQ0P-ayeiTeKRgAJR0El4B8vzBQCsjjNMjkwoqwJLLjaL3G6Tt6uGJq4EoeAADlwgeAdKH%2BZJOAeexbW6LqxONPrIAecJZGsLTtEISgnysWl2Oowaz1kxcQMmlqBgOo6JlO87dOTaA5kpItaQAfRlJhfIWLaQvE3aIqkqLBW6S9xqUyg2EwX5KHGAs2oWfyiuEj9ay-SHKpuhjrmAttBhGRHkdR9H0DmhbaWW1b1thsHStC8q9pephjsQd6ODvK7fGJmTYiWbDyaplG0aYAtnsOnm3pJD7By%2Bn6kb%2ByhAeB-i2fxsq4ShnZquG%2BglhucaADVT3xER4FsTHS06oKEPBnbOaNmETcFJZ5PJ%2Bz8FUos8tttIHZ1rGOpxwy8cojnDaJyKhp92zxrU2B0AAcUsTLsv0PLHYK6OyljkyCc5yTjekmrrji-3ud5-FXmgABrfENZbyx1DYjIqTXI55vy2Dtz1uOIYr38q5htsljq8mZsQGVG8sOVpiFxANKSmg%2BLy940j7BnKBZts4Jd0S3frCTJ696vTYgJYxvJgBpRBEGwRznOmVJe%2BpQE8mHwSo8y4Gz8J7EI3sZ73X9iMNyTx1BzELlHIBPVJwoR5EnW6Yt4bkyyDMWQiAlI2HmtMbyiApADAQP-E%2BI8z7dXLlya%2B4Db6CnRGTAokB3haRsDQQONBg5vCyNwjg4RSTsVgMxEhHA8oCQrLjEq%2Bt440XQdDZObZ0QS3YZwu8NApEsEmG8GwB9FqbxEDvfMMiDIl3kWPd2DDaIYJJrEdEFt64Kz-qMKkUh4BsCzLAXKMwF5LwmBY5B9ClFVWYWov27C1KyzYFIfBsDZZzAyJgUQIwQm0O2pfXa0x4D-D-hkLA%2BCInTwKOiVO5Mfj5IBMEr6lBZACEgKEkBR57EqMwe4dEdd2EvzfhI0QqR25sD4ZHahgCskXzMrtSuN8ylJHRHPdhRTmBFgoUgTJxV9RmFKPkfw6h8C5QcHBCoQA))

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8c09996f-24ac-cf39-ef43-d7909cbc2b89
* Version Independent ID: 1efdd523-8552-6421-e149-74f76826ef8f
* Content: [F# code formatting guidelines](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting)
* Content Source: [docs/fsharp/style-guide/formatting.md](https://github.com/dotnet/docs/blob/main/docs/fsharp/style-guide/formatting.md)
* Product: **dotnet-fsharp**
* GitHub Login: @cartermp
* Microsoft Alias: **phcart**

### PR DESCRIPTION
**Pull request by nojaf**
Fixes #23581

### CODE REVIEW

File: docs/fsharp/style-guide/formatting.md [hunk 1]
```diff
@@ -556,6 +556,38 @@ type Volume =
 | ImperialPint of float
 ```
 
+If there is a single short union you can omit the leading `|`.
```
Comment (Youssef1313) [inline docs/fsharp/style-guide/formatting.md:559]:
```suggestion
If there is a single short union, you can omit the leading `|`.
```

File: docs/fsharp/style-guide/formatting.md [hunk 2]
```diff
@@ -556,6 +556,38 @@ type Volume =
...
+```fsharp
+type Address = Address of string
+```
+
+For a longer or multiline union, keep the `|`.
+
+```fsharp
+[<NoEquality; NoComparison>]
+type SynBinding =
+    SynBinding of
```
Comment (IEvangelist) [inline docs/fsharp/style-guide/formatting.md:570]:
```suggestion
    | SynBinding of
```

File: docs/fsharp/style-guide/formatting.md [hunk 3]
```diff
@@ -556,6 +556,38 @@ type Volume =
 | ImperialPint of float
 ```
 
+If there is a single short union, you can omit the leading `|`.
```
Comment (IEvangelist) [inline docs/fsharp/style-guide/formatting.md:559]:
```suggestion
When there is a single short union, you can omit the leading `|`.
```

File: docs/fsharp/style-guide/formatting.md [hunk 4]
```diff
@@ -556,6 +556,38 @@ type Volume =
...
+        attributes: SynAttributes *
+        xmlDoc: PreXmlDoc *
+        valData: SynValData *
+        headPat: SynPat *
+        returnInfo: SynBindingReturnInfo option *
+        expr: SynExpr *
+        range: range *
+        seqPoint: DebugPointAtBinding
+```
+
```
Comment (IEvangelist) [inline docs/fsharp/style-guide/formatting.md:584]:
```suggestion

You can also use triple-slash `///` comments.

```

**Review by IEvangelist — APPROVED** [review]
This looks good, thank you. We'll :shipit: when you make the suggested changes.

File: docs/fsharp/style-guide/formatting.md (hunk 2 above)
Comment (nojaf) [reply docs/fsharp/style-guide/formatting.md:570]: Good spot! That was exactly the point of adding that sample 😅.

**Review by cartermp — APPROVED** [review]
Thanks!

**Pull request merged**